// @yext/search-core/dist/commonjs/src/transformers/autocompleteservice/createAutocompleteResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAutocompleteResult = void 0;
const Source_1 = require("../../models/searchservice/response/Source");
const createFieldValueFilter_1 = require("../core/createFieldValueFilter");
const ResultsFactory_1 = require("../searchservice/ResultsFactory");
function createAutocompleteResult(result) {
    var _a;
    const relatedItem = result.relatedItem
        ? ResultsFactory_1.ResultsFactory.create([result.relatedItem], Source_1.Source.KnowledgeManager)[0]
        : result.relatedItem;
    return {
        filter: result.filter && (0, createFieldValueFilter_1.createFieldValueFilter)(result.filter),
        key: result.key,
        matchedSubstrings: result.matchedSubstrings || [],
        value: result.value,
        relatedItem: relatedItem,
        verticalKeys: result.verticalKeys,
        inputIntents: (_a = result.queryIntents) !== null && _a !== void 0 ? _a : []
    };
}
exports.createAutocompleteResult = createAutocompleteResult;
