// @yext/answers-core/dist/commonjs/src/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SandboxEndpoints = exports.defaultEndpoints = exports.defaultApiVersion = void 0;
exports.defaultApiVersion = 20220511;
exports.defaultEndpoints = {
    universalSearch: 'https://liveapi.yext.com/v2/accounts/me/answers/query',
    verticalSearch: 'https://liveapi.yext.com/v2/accounts/me/answers/vertical/query',
    questionSubmission: 'https://liveapi.yext.com/v2/accounts/me/createQuestion',
    status: 'https://answersstatus.pagescdn.com',
    universalAutocomplete: 'https://liveapi-cached.yext.com/v2/accounts/me/answers/autocomplete',
    verticalAutocomplete: 'https://liveapi-cached.yext.com/v2/accounts/me/answers/vertical/autocomplete',
    filterSearch: 'https://liveapi-cached.yext.com/v2/accounts/me/answers/filtersearch',
};
/**
 * The endpoints to use for sandbox experiences.
 *
 * @public
 */
exports.SandboxEndpoints = {
    universalSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/answers/query',
    verticalSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/answers/vertical/query',
    questionSubmission: 'https://liveapi-sandbox.yext.com/v2/accounts/me/createQuestion',
    status: 'https://answersstatus.pagescdn.com',
    universalAutocomplete: 'https://liveapi-sandbox.yext.com/v2/accounts/me/answers/autocomplete',
    verticalAutocomplete: 'https://liveapi-sandbox.yext.com/v2/accounts/me/answers/vertical/autocomplete',
    filterSearch: 'https://liveapi-sandbox.yext.com/v2/accounts/me/answers/filtersearch',
};
